elyra/pipeline/validate.py

```diff
@@ -0,0 +1,552 @@
+            components = json.loads(ComponentRegistry.to_canvas_palette(component_list))
+            for category in components['categories']:
+                supported_ops.append(category['node_types'][0]['op'])
+
+            # Checks pipeline node types are compatible with the runtime selected
+            for single_pipeline in pipeline_json['pipelines']:
+                pipeline_id = single_pipeline['id']
+                node_list = single_pipeline['nodes']
+                for node in node_list:
+                    if node['type'] == "execution_node" and node['op'] not in supported_ops:
+                        response.add_message(severity=ValidationSeverity.Error,
+                                             message_type="invalidNodeType",
+                                             message="Unsupported node type found in this pipeline",
+                                             data={"nodeID": node['id'],
+                                                   "nodeOpName": node['op'],
+                                                   "nodeName": node['app_data']['ui_data']['label'],
+                                                   "pipelineID": pipeline_id})
+        else:
+            response.add_message(severity=ValidationSeverity.Error,
+                                 message_type="invalidRuntime",
+                                 message="Unsupported pipeline runtime selected in this pipeline",
+                                 data={"pipelineRuntime": pipeline_runtime,
+                                       "pipelineID": pipeline_id})
+
+    async def _validate_node_properties(self, root_dir: str, pipeline: Dict, pipeline_runtime: str,
+                                        pipeline_execution: str, response: ValidationResponse) -> None:
+        """
+        Validates each of the node's structure for required fields/properties as well as
+        their values
+        :param root_dir: the absolute base path of the current elyra workspace
+        :param pipeline: the pipeline definition to be validated
+        :param pipeline_runtime: name of the pipeline runtime being used e.g. kfp, airflow, generic
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        pipeline_json = json.loads(json.dumps(pipeline))
+
+        for single_pipeline in pipeline_json['pipelines']:
+            node_list = single_pipeline['nodes']
+            pipeline_runtime = 'local' if pipeline_runtime == 'generic' else pipeline_runtime
+            component_list = await PipelineProcessorManager.instance().get_components(pipeline_runtime)
+            components = json.loads(ComponentRegistry.to_canvas_palette(component_list))
+            for node in node_list:
+                if node['type'] == 'execution_node':
+                    node_data = node['app_data']
+                    if node['op'] in ['execute-r-node', 'execute-python-node', 'execute-notebook-node']:
```

Comment:
We should add a class method on Operator that takes the `node['op']` value and returns true if `isGenericOperator()`.